config/snapshot/config_snapshot.go

```diff
@@ -29,130 +28,67 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// Config is a mapped representation of the config at a particular point in time
-type Config struct {
-	dashboards map[string]*configpb.Dashboard
-	groups     map[string]*configpb.TestGroup
-	attrs      storage.ReaderObjectAttrs
-	lock       sync.RWMutex
+// ConfigSnapshot is a mapped representation of the config at a particular point in time.
+// Not concurrency-safe; meant for reading only
+type ConfigSnapshot struct {
+	Dashboards map[string]*configpb.Dashboard
+	Groups     map[string]*configpb.TestGroup
+	Attrs      storage.ReaderObjectAttrs
 }
 
-// Dashboard returns a dashboard's configuration
-func (cs *Config) Dashboard(name string) *configpb.Dashboard {
-	if cs == nil {
-		return nil
+// Observe reads the config at configPath and return a ConfigSnapshot initially and, if a ticker is supplied, when the config changes
+// Returns an error instead if there is no file at configPath
+func Observe(ctx context.Context, log logrus.FieldLogger, client gcs.ConditionalClient, configPath gcs.Path, ticker <-chan time.Time) (<-chan ConfigSnapshot, error) {
+	ch := make(chan ConfigSnapshot)
```

Comment:
Any reason you are not doing `chan *ConfigSnapshot`?

Reply:
Didn't need it to be nil-able, and I don't know what other meaningful difference there is between "snapshot.Config, a struct of pointers" and "*snapshot.Config, a pointer to a struct of pointers".

I'm likely to switch this if I switch the signature for `updateHash`; being consistent throughout the code is a pretty good reason.